shared/cert.go

```diff
@@ -245,23 +248,28 @@ func TestingAltKeyPair() *CertInfo {
 	return cert
 }
 
-/*
- * Generate a list of names for which the certificate will be valid.
- * This will include the hostname and ip address.
- * If the `name` argument is non-empty,  it will be used in place of the system hostname.
- */
-func mynames(name string) ([]string, error) {
-	if name == "" {
+// generateSANNames creates a list of names for which the certificate will be valid.
+// - `commonName` will be the first entry if defined, otherwise the hostname will be used.
+// - `additionalNames` will be supplied next, if defined
+// - Finally, the localhost IPs will be added.
+func generateSANNames(commonName string, additionalNames []string) ([]string, error) {
```

Comment:
Can you make additionalNames variadic please?

Reply:
Done. 

I thought about removing `commonName` as an argument and defaulting to the hostname if `additionalNames` is empty, but that doesn't actually reduce any complexity since we'd still have to append the hostname to the args passed into `generateSANNames` anyway.